src/old-ecs/core-ecs/core-ecs-types.ts

```diff
@@ -159,6 +159,19 @@ export interface Table<C = { [K: string]: unknown }> {
   readonly columns: { [K in keyof C]: Column<C[K]> };
 }
 
+/**
+ * A set of flags that can be used to control the privacy of the data.
+ */
+export interface UserPrivacyPreferences {
+/**
+ * Allows unsafe direct access to underlying attributes.
+ */
+  strictlyNecessary: boolean;
+  performance: boolean;
+  functional: boolean;
+  advertising: boolean;
+}
+
```

Comment:
Don't add this type to core definitions. This will only be necessary as a parameter passed into `toJSON` or whichever function is performing the serialization.